### PR TITLE
Print deployment status after every 0.5 seconds.

### PR DESCRIPTION
**Pull request by tejal29**
relates to #176
Should merge after #2865 

In this PR, after ever **0.5 seconds**, we will printing the deployment status to give more information to user on what the current status of the check is.
Let me know if 0.5 seconds is too frequent and we shd bump it to a 1 second.

**Output Before:**
```
$ skaffold run --default-repo gcr.io/tejal-test --status-check

Waiting for deployments to stabilize
 - default:deployment/leeroy-web is ready. [1/2 deployment(s) still pending]
 - default:deployment/leeroy-app is ready.
Deployments stabilized in 2.177879554s
```

**Output After:**
```
$ skaffold run --default-repo gcr.io/tejal-test --status-check
...
Deploy complete in 1.159729575s
Waiting for deployments to stabilize
 - default:deployment/leeroy-app Waiting for rollout to finish: 0 of 1 updated replicas are available...
 - default:deployment/leeroy-web Waiting for rollout to finish: 0 of 1 updated replicas are available...
 - default:deployment/leeroy-app is ready. [1/2 deployment(s) still pending]
 - default:deployment/leeroy-web is ready.
Deployments stabilized in 1.766328247s
```

Output for different scenarios:

1. examples/microservices with increased replicas.
```
$ skaffold run --default-repo gcr.io/tejal-test --status-check
...
Deploy complete in 1.083255687s
Waiting for deployments to stabilize
 - default:deployment/leeroy-app Waiting for rollout to finish: 0 of 2 updated replicas are available...
 - default:deployment/leeroy-web Waiting for rollout to finish: 0 of 3 updated replicas are available...
 - default:deployment/leeroy-web is ready. [1/2 deployment(s) still pending]
 - default:deployment/leeroy-app is ready.
Deployments stabilized in 4.669434037s


```

2.  Hitting Cntrl-C 
```
$ skaffold run --default-repo gcr.io/tejal-test --status-check
...
Deploy complete in 1.105834424s
Waiting for deployments to stabilize
 - default:deployment/leeroy-app Waiting for rollout to finish: 3 out of 5 new replicas have been updated...
 - default:deployment/leeroy-web Waiting for rollout to finish: 1 out of 3 new replicas have been updated...
^C - default:deployment/leeroy-web failed. [1/2 deployment(s) still pending] Error: Running [kubectl --context gke_tejal-test_us-central1-a_dump rollout status deployment leeroy-web --watch=false]: stdout , stderr: , err: signal: interrupt: signal: interrupt.
 - default:deployment/leeroy-app failed. Error: Running [kubectl --context gke_tejal-test_us-central1-a_dump rollout status deployment leeroy-app --watch=false]: stdout , stderr: , err: signal: interrupt: signal: interrupt.
```

### CODE REVIEW

File: pkg/skaffold/deploy/resource/deployment.go [hunk 1]
```diff
@@ -79,3 +88,9 @@ func (d *Deployment) WithStatus(details string, err error) *Deployment {
 	d.UpdateStatus(details, err)
 	return d
 }
+
+func (d *Deployment) WithDone(details string, err error) *Deployment {
```
Comment (priyawadhwa) [inline pkg/skaffold/deploy/resource/deployment.go]: Is this function only used for unit tests?

If so, maybe we could move it to the test file as a helper function --


```go
func WithDone(d *Deployment, details string, err error) *Deployment
```

File: pkg/skaffold/deploy/status_check.go [hunk 2]
```diff
@@ -171,6 +180,38 @@ func printStatusCheckSummary(d *resource.Deployment, pending int, total int, err
 	color.Default.Fprintln(out, status)
 }
 
+// Print resource statuses until all status check are completed or context is cancelled.
+func printResourceStatus(ctx context.Context, deps []*resource.Deployment, deadline time.Duration, out io.Writer) {
```
Comment (nkubala) [inline pkg/skaffold/deploy/status_check.go]: nit: usually we do the context, then the writer, then the other stuff.

`func printResourceStatus(ctx context.Context, out io.Writer, deps []*resource.Deployment, deadline time.Duration)`

File: pkg/skaffold/deploy/resource/deployment.go (hunk 1 above)
Comment (tejal29) [reply pkg/skaffold/deploy/resource/deployment.go]: sounds good.

File: pkg/skaffold/deploy/status_check.go (hunk 2 above)
Comment (tejal29) [reply pkg/skaffold/deploy/status_check.go]: sounds good.

File: pkg/skaffold/deploy/status_check_test.go [hunk 3]
```diff
@@ -430,8 +439,11 @@ func TestPrintStatus(t *testing.T) {
...
 			rs: []*resource.Deployment{
-				resource.NewDeployment("r1", "test", 1).
-					WithDone("succes", nil),
+				withDone(
+					resource.NewDeployment("r1", "test", 1),
+					"succes",
+					nil,
+				),
 				resource.NewDeployment("r2", "test", 1).
 					WithStatus("", fmt.Errorf("context deadline expired")),
```
Comment (priyawadhwa) [inline pkg/skaffold/deploy/status_check_test.go]: Looks like `WithStatus` can be moved into the unit test as a helper function as well?

File: pkg/skaffold/deploy/status_check_test.go [hunk 4]
```diff
@@ -404,24 +404,33 @@ func TestPrintStatus(t *testing.T) {
...
 			expected: true,
 		},
 		{
 			description: "multiple resources 1 not complete",
 			rs: []*resource.Deployment{
-				resource.NewDeployment("r1", "test", 1).
-					WithDone("succes", nil),
+				withDone(
+					resource.NewDeployment("r1", "test", 1),
+					"succes",
```
Comment (priyawadhwa) [inline pkg/skaffold/deploy/status_check_test.go]: nit: success

File: pkg/skaffold/deploy/status_check_test.go [hunk 5]
```diff
@@ -430,8 +439,11 @@ func TestPrintStatus(t *testing.T) {
 		{
 			description: "multiple resources 1 not complete and in error",
 			rs: []*resource.Deployment{
-				resource.NewDeployment("r1", "test", 1).
-					WithDone("succes", nil),
+				withDone(
+					resource.NewDeployment("r1", "test", 1),
+					"succes",
```
Comment (priyawadhwa) [inline pkg/skaffold/deploy/status_check_test.go]: nit: success

File: pkg/skaffold/deploy/status_check_test.go (hunk 3 above)
Comment (tejal29) [reply pkg/skaffold/deploy/status_check_test.go]: yes. i think i will do that in another PR.

**Pull request merged**